### DISCUSSION TITLE
How to set Arbitrary values if both border width and border color are used simultaneously

### DISCUSSION THREAD

**Discussion by yudan215** [tailwindlabs/tailwindcss#12118] [Help] [answered]
How to set Arbitrary values if both border width and border color are used simultaneously

**Comment by wongjn** [comment]
Arbitrary value class for which group of classes/CSS property please? Consider providing a [Tailwind Play](https://play.tailwindcss.com) that displays your problem.

**Reply by yudan215** [reply]
Class='border - [hsla (0,0%, 100%,. 05)] border - [8px]' This way, border - [8px] will be invalid

**Reply by wongjn** [reply]
You have spaces in your classes which means Tailwind won't see these classes. Instead of `border - [hsla (0,0%, 100%,. 05)] border - [8px]`, consider `border-[hsla(0,0%,100%,.05)] border-[8px]`. See [this Tailwind Play](https://play.tailwindcss.com/0kWkXYMXHh) of it working (open the *Generated CSS* panel).

**Reply by yudan215** [reply]
There are actually no spaces, maybe I just typed too much here

**Reply by wongjn** [reply]
There does seem to spaces in your reply:
![image](https://github.com/tailwindlabs/tailwindcss/assets/11310624/e4f6971e-a0e8-4984-a406-c7ee8f4a8439)

See [this Tailwind Play](https://play.tailwindcss.com/0kWkXYMXHh) of it working without spaces (open the *Generated CSS* panel).

**Reply by yudan215** [reply]
border-[hsla(0,0%,100%,.05)] border-[8rpx] 这个才是

**Reply by yudan215** [reply]
But border-[8rpx] will be compiled as border color, resulting in invalidity

**Reply by wongjn** [reply]
`rpx` is an invalid CSS unit, so Tailwind then assumes its a color. You may have meant `px` unit, like `border-[8px]` instead of `border-[8rpx]`. Furthermore, as per [the documentation](https://tailwindcss.com/docs/border-width) with default configuration, `border-width: 8px` can be applied using a built-in class `border-8` so an arbitrary value class may not be needed.

**Reply by yudan215** [reply]
The RPX unit is because I am using it in WeChat mini programs, which has nothing to do with the unit. The main thing is that it will be compiled into a border-color

**Reply by wongjn** [reply, marked as answer]
You can disambiguate the arbitrary value class by [type-hinting](https://tailwindcss.com/docs/adding-custom-styles#resolving-ambiguities) the class like `border-[length:8rpx]`. See [this Tailwind Play](https://play.tailwindcss.com/DLjrxsqpRu) of it working (open the *Generated CSS* panel).

**Reply by yudan215** [reply]
thanks
